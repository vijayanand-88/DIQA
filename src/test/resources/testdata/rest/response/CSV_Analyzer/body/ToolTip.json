{
  "configurationName": null,
  "configurationType": "plugin-configuration",
  "type": "Dataanalyzer",
  "pluginName": "CsvAnalyzer"
}