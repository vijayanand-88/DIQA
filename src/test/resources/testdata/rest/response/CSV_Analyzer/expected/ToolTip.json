{
  "Commonfields": {
    "Name": {
      "name": "name",
      "widget": "string",
      "label": "Name",
      "tooltip": "Plugin configuration name",
      "hidden": false,
      "validators": [
        {
          "name": "required",
          "message": " Name field should not be empty"
        },
        {
          "name": "uniqueTableKey",
          "message": " Name contains duplicated values"
        },
        {
          "name": "pattern",
          "config": "^[^\\s\\/\\\\]+([\\s]+[^\\s\\/\\\\]+)*$",
          "message": " Invalid name. Leading/trailing blanks and special characters are forbidden"
        },
        {
          "name": "uniqueValue",
          "config": [
            "MongoDBCataloger_ValidDB&Cred"
          ],
          "message": "Name already exists. Please enter a different name."
        }
      ],
      "advanced": false
    },
    "type": {
      "name": "type",
      "widget": "select",
      "label": "Type",
      "tooltip": "Type of the plugin",
      "hidden": false,
      "validators": [
        {
          "name": "requiredSelection",
          "message": " Type should have at least one value selected"
        }
      ],
      "advanced": false
    },
    "pluginName": {
      "name": "pluginName",
      "widget": "select",
      "label": "Plugin",
      "tooltip": "Name of the plugin",
      "hidden": false,
      "validators": [
        {
          "name": "requiredSelection",
          "message": " Plugin should have at least one value selected"
        }
      ],
      "advanced": false
    },
    "pluginVersion": {
      "name": "pluginVersion",
      "widget": "select",
      "label": "Plugin version",
      "tooltip": "Required plugin version",
      "hidden": false,
      "validators": [
      ],
      "advanced": false
    },
    "label": {
      "value": null,
      "name": "label",
      "widget": "label",
      "label": "Label",
      "tooltip": "Plugin configuration extended label and description",
      "hidden": false,
      "validators": [
      ],
      "advanced": false
    },
    "businessApplicationName": {
      "name": "businessApplicationName",
      "widget": "select",
      "label": "Business Application",
      "tooltip": "Business Application",
      "hidden": false,
      "validators": [
      ],
      "advanced": false
    },
    "dataSource": {
      "name": "dataSource",
      "widget": "childProperties",
      "label": "Data Source",
      "tooltip": "Data source connection to be used",
      "hidden": false,
      "validators": [
      ],
      "advanced": false
    },
    "credential": {
      "name": "credential",
      "widget": "childProperties",
      "label": "Credential",
      "tooltip": "Credential to be used",
      "hidden": false,
      "validators": [
      ],
      "advanced": false
    },
    "eventCondition": {
      "value": null,
      "name": "eventCondition",
      "widget": "string",
      "label": "Event condition",
      "tooltip": "Condition an event must match to trigger this configuration. Expressed in Groovy",
      "hidden": false,
      "validators": [
      ],
      "advanced": true
    },
    "dryRun": {
      "value": false,
      "name": "dryRun",
      "widget": "checkbox",
      "label": "Dry Run",
      "tooltip": "Option to enable/disable the plugin execution on dry run mode without writing objects into the repository",
      "hidden": false,
      "validators": [
      ],
      "advanced": true
    },
    "eventClass": {
      "value": null,
      "name": "eventClass",
      "widget": "string",
      "label": "Event class",
      "tooltip": "Event Class name to trigger current plugin configuration",
      "hidden": false,
      "validators": [
      ],
      "advanced": true
    },
    "maxWorkSize": {
      "value": 100,
      "name": "maxWorkSize",
      "widget": "number",
      "label": "Maximum work size",
      "tooltip": "Maximum number of items the plugin can process before delegating to other nodes",
      "hidden": false,
      "validators": [
      ],
      "advanced": true
    },
    "nodeCondition": {
      "value": null,
      "name": "nodeCondition",
      "widget": "string",
      "label": "Node condition",
      "tooltip": "Condition an node must match to be able to run this configuration. Expressed in Groovy",
      "hidden": false,
      "validators": [
      ],
      "advanced": true
    },
    "autoStart": {
      "name": "autoStart",
      "widget": "checkbox",
      "label": "Auto start",
      "tooltip": "Option to enable/disable the automatic execution of the plugin configuration after node startup",
      "hidden": false,
      "validators": [
      ],
      "advanced": true
    },
    "tags": {
      "name": "tags",
      "widget": "multi-select",
      "label": "Tags",
      "tooltip": "Add custom tag(s) to define your data item",
      "hidden": false,
      "validators": [
      ],
      "advanced": true
    },
    "runAfter": {
      "name": "runAfter",
      "widget": "table",
      "label": "Run automatically after",
      "tooltip": "Plugin configurations to run after. Any matching configuration will trigger this one.",
      "hidden": false,
      "validators": [],
      "advanced": true
    },
    "schedule": {
      "name": "schedule",
      "widget": "schedule",
      "label": "Schedule",
      "tooltip": "Run plugin automatically in this configuration at regular intervals",
      "hidden": false,
      "validators": [],
      "advanced": true
    }
  },
  "Uniquefilter": {
    "CSVAnlayzer": {
      "sampleSize": {
        "name": "sampleSize",
        "widget": "number",
        "label": "Sample size",
        "tooltip": "Number of rows that would be retrieved from csv file.Range is between 10-1000.",
        "hidden": false,
        "validators": [
          {
            "name": "required",
            "message": " Sample size field should not be empty"
          }
        ],
        "advanced": false
      },
      "histogramBuckets": {
        "name": "histogramBuckets",
        "widget": "number",
        "label": "Histogram buckets",
        "tooltip": "Number of buckets for representing data distribution. Range is between 5-20.",
        "hidden": false,
        "validators": [
          {
            "name": "required",
            "message": " Histogram buckets field should not be empty"
          }
        ],
        "advanced": false
      },
      "topValues": {
        "name": "topValues",
        "widget": "number",
        "label": "Top values",
        "tooltip": "Most popular values in csv files.Range is between 5-30.",
        "hidden": false,
        "validators": [
          {
            "name": "required",
            "message": "Most popular values in csv files.Range is between 5-30."
          }
        ],
        "advanced": false
      }
    }
  }
}